ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,353 @@
+                    break;
+                case SortBy.Phone:
+                    for (int i = 0; i < array.Length; i++)
+                    {
+                        for (int j = 0; j < array.Length - 1; j++)
+                        {
+                            if ((array[j].phone.CompareTo(array[j + 1].phone) >= 1 && order == OrderBy.Ascending) ||
+                                (array[j].phone.CompareTo(array[j + 1].phone) <= -1 && order == OrderBy.Descending))
+                            {
+                                (string name, string surname, string phone) t = array[j];
+                                array[j] = array[j + 1];
+                                array[j + 1] = t;
+                            }
+
+                        }
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        static bool BinarySearch(string path, out int id)
+        {
+            // get the info to search
+            Console.WriteLine("Enter surname to search: ");
+            string search = Console.ReadLine() ?? "";
+            // get the info u need 
+            (string firstname, string lastname, string phone)[] records;
+            SortBinary(path);
+            records = ReadPhoneBookRecords(path);
+            // sort at first
+
+            // search 
+            int l = 0;
+            int r = records.Length - 1;
+            int mid;
+            while (l <= r)
+            {
+                mid = (l + r) / 2;
+                if (search == records[mid].lastname)
+                {
+                    id = mid;
+                    return true;
+                }
+                else if (search.CompareTo(records[mid].lastname) <= -1)
+                {
+                    r = mid - 1;
+                }
+                else
+                {
+                    l = mid + 1;
+                }
+            }
+            id = -1;
+            return false;
+        }
+
+        static void EditRecord(string path)
+        {
+
+            var records = File.ReadAllLines(path);
+            int input = IdInput(records.Length);
+
+            Console.WriteLine("Founded record: ");
+            Console.WriteLine(records[input]);
+
+            Console.Write("Enter new firstname: ");
+            string newName = Console.ReadLine() ?? "";
+
+            Console.Write("Enter new lastname: ");
+            string newLast = Console.ReadLine() ?? "";
+
+            Console.Write("Enter new phone: ");
+            string newPhone = Console.ReadLine() ?? "";
+
+            string newRecord = string.Join(',', newName, newLast, newPhone);
+            records[input] = newRecord;
+            File.WriteAllLines(path, records);
+        }
+
+        static void DeleteRecord(string path)
+        {
+            var records = File.ReadAllLines(path);
+            int input = IdInput(records.Length);
+
+            DeleteSwap(records, input);
```

Comment:
Delete operation change order of elements after delete

Reply:
Changed the way of deleting elements.